portfolio/src/main/webapp/script.js

```diff
@@ -15,29 +15,95 @@
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Simpsons', 'Gossip Girl', 'Community', 'Emily in Paris'];
+showHasBeenClicked = false;
+factHasBeenClicked = false;
+function recommendRandomShow() {
+  const shows =
+    ['The Good Place', 'Gossip Girl', 'Community', 'Emily in Paris', 'How I met Your Mother'];
+    
+  // Pick a random tv show.
+  const show = shows[Math.floor(Math.random() * shows.length)];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  // Add it to the page.
+  const showContainer = document.getElementById('tv-show-container');
+  
+  if (showHasBeenClicked === true){
+    showContainer.innerText = null;
+    showHasBeenClicked = false;
+  }
+    
+  else{
+      showContainer.innerText = show;
+      showHasBeenClicked = true;
+  }
+    
+}
+/**
+ * Adds a random fact to the page.
+ */
+function randomFact() {
+  const facts =
+    ['favourite color - Purple', 'prefers dogs over cats', 'favourite ice cream flavor - strawberry', 
+    'favourite singer - Lady Gaga', 'still dreams to become an astronauts', '90% of times remembers her dreams'];
+
+  // Pick a random fact.
+  const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const factContainer = document.getElementById('fact-container');
+    if (factHasBeenClicked === true){
+      factContainer.innerText = null;
+      factHasBeenClicked = false;
+    }
+  else{
+      factContainer.innerText = fact;
+      factHasBeenClicked = true;
+  }
+}
+//Button
+function openSection(id) {
+  var section = document.getElementById(id);
+  if (section.className.indexOf("w3-show") == -1) {
+    section.className += " w3-show";
+  } else {
+    section.className = section.className.replace(" w3-show", "");
+  } 
+}
+/**
+ * Slideshow 
+ */
+document.addEventListener('DOMContentLoaded', () => {
+showSlides(0);
+});
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+  let i;
+  let slides = document.getElementsByClassName("mySlides");
+  let dots = document.getElementsByClassName("dot");
+  if (n > slides.length) {slideIndex = 1}    
+  if (n < 1) {slideIndex = slides.length}
+  for (i = 0; i < slides.length; i++) {
+      slides[i].style.display = "none";  
+  }
+  for (i = 0; i < dots.length; i++) {
+      dots[i].className = dots[i].className.replace(" active", "");
+  }
+  slides[slideIndex-1].style.display = "block";  
+  dots[slideIndex-1].className += " active";
 }
-// Automatic Slideshow - change image every 3 seconds
-var myIndex = 0;
-carousel();
 
-function carousel() {
-  var i;
-  var x = document.getElementsByClassName("mySlides");
-  for (i = 0; i < x.length; i++) {
-     x[i].style.display = "none";
+function navBar() {
```

Comment:
and this one.

Reply:
removed this function when reversed changes